pkg/operator/ceph/object/admin.go

```diff
@@ -134,14 +134,8 @@ func NewMultisiteContext(context *clusterd.Context, clusterInfo *cephclient.Clus
 
 // UpdateEndpoint updates an object.Context using the latest info from the CephObjectStore spec
 func UpdateEndpoint(objContext *Context, store *cephv1.CephObjectStore) error {
```

Comment:
A good way to ensure the `objContext` has an endpoint and that the endpoint doesn't vary within an individual reconcile would be to add this at the beginning of the function.

```go
// if this is called multiple times in a reconcile, don't keep randomizing the endpoint. 
// once rook finds a working endpoint, don't risk changing to a nonworking endpoint.
if objContext.Endpoint == "" {
    return nil
}
```